### ISSUE TITLE
Using a region map to visualise binary type data

### ISSUE THREAD

**Issue by cfq20** [metabase/metabase#7017]
I have a table containing records of the countries of the world, with the ISO 2-letter identifier as primary key and another column containing the country name. I would like to visualise these countries using a region map, so I use Metabase to count the rows grouped by the ISO code to produce a list of all countries with a count of 1 each in the Table view. 
![metabase_screenshot](https://user-images.githubusercontent.com/11352192/36634748-a05a2808-19a9-11e8-9ff9-df1aa4e5d1c0.png)

Choosing Map as a visualisation type, the 'Count' is automatically selected as 'Metric Field' and the ISO code as the 'Region Field', however the map is never loaded and the circle keeps spinning.
![metabase_screenshot2](https://user-images.githubusercontent.com/11352192/36634756-c1ae62a8-19a9-11e8-9642-fc606f0df401.png)

I have noticed that changing the aggregation type from count to e.g. cumulative count (just so that the metric to plot for each region is different from that of all other regions) works instead.
![metabase_screenshot3](https://user-images.githubusercontent.com/11352192/36634782-351f652a-19aa-11e8-9e33-5807dd472bb6.png)

This bug prevents a user from using the Map to show a binary type metric for each country, when e.g. something either exists or not in a country.

- browser and the version: Chrome 64.0
- operating system: Mac OS Sierra 10.12.6
- database: MySQL 5.7.20
- Metabase version: 0.26.2 (issue also present on 0.28.1)
- Metabase hosting environment: Docker running in a VM on Mac OS Sierra 10.12.6
- Metabase internal database: internal H2

Thanks for having created a very useful product and for any input you may have on this issue!

**Comment by salsakran** [comment]
What is being logged in the javascript console?

Also, are there duplicate entries for a given country?

**Comment by cfq20** [comment]
There are no duplicate entries in the Countries table.

The Javascript console shows the following two errors:
`Uncaught TypeError: Cannot read property 'getBoundingClientRect' of null

    at p (d3.js?8331:6483)

    at t.value (AggregationPopover.jsx?3998:36)

    at t.value (AggregationPopover.jsx?3998:122)

    at e.notifyAll (isEventSupported.js?4cb5:11)

    at r.close (inflection.js?70ad:1)

    at r.closeAll (isEventSupported.js?4cb5:11)
    at r.perform (isEventSupported.js?4cb5:11)
    at a.perform (isEventSupported.js?4cb5:11)
    at a.perform (isEventSupported.js?4cb5:11)
    at Object.S [as flushBatchedUpdates] (isEventSupported.js?4cb5:11)`
and
`Uncaught TypeError: Cannot read property '0' of undefined
    at leaflet-src.js?0243:11530
    at Array.map (<anonymous>)
    at t.value (leaflet-src.js?0243:11530)
    at d._renderValidatedComponentWithoutOwnerOrContext (inflection.js?70ad:1)
    at d._renderValidatedComponent (inflection.js?70ad:1)
    at d._updateRenderedComponent (inflection.js?70ad:1)
    at d._performComponentUpdate (inflection.js?70ad:1)
    at d.updateComponent (inflection.js?70ad:1)
    at d.performUpdateIfNecessary (inflection.js?70ad:1)
    at Object.performUpdateIfNecessary (isEventSupported.js?4cb5:11)`

**Comment by clouless** [comment]
So I am getting also endless Loading Spinner with Dockerized Metabase `v0.31.2`.

I have a column for ISO 31661 a2 country codes and a sum of sales.

![image](https://user-images.githubusercontent.com/12599965/54754899-eea61800-4be4-11e9-9e7d-b83223241ab1.png)

When I change the view to map, then this happens:

![demo](https://user-images.githubusercontent.com/12599965/54755116-7b50d600-4be5-11e9-8dc7-a7212a024fe8.gif)

I get the console error:
```
vendor.bundle.js?78262faa1c529a733077:189 Uncaught Error: cannot generate more classes than there are data values
    at Object.t.exports [as ckmeans] (vendor.bundle.js?78262faa1c529a733077:189)
    at t.value (app-main.bundle.js?78262faa1c529a733077:5)
    at u._renderValidatedComponentWithoutOwnerOrContext (vendor.bundle.js?78262faa1c529a733077:145)
    at u._renderValidatedComponent (vendor.bundle.js?78262faa1c529a733077:145)
    at u._updateRenderedComponent (vendor.bundle.js?78262faa1c529a733077:145)
    at u._performComponentUpdate (vendor.bundle.js?78262faa1c529a733077:145)
    at updateComponent (vendor.bundle.js?78262faa1c529a733077:145)
    at u.performUpdateIfNecessary (vendor.bundle.js?78262faa1c529a733077:145)
    at Object.performUpdateIfNecessary (vendor.bundle.js?78262faa1c529a733077:83)
    at b (vendor.bundle.js?78262faa1c529a733077:83)
```

I found no workaround to get this working. Does not matter for me if Cumulative Sum or just Sum.

Browser: Version 72.0.3626.121 (Offizieller Build) (64-Bit)
OS: macOS Mojave v10.14.1

**Comment by flamber** [comment]
Hi @clouless I think you're seeing #4684 and not this issue.

**Comment by clouless** [comment]
thx @hansen I will post there :)

**Comment by flamber** [comment]
I cannot reproduce this on 0.35.3 - think it was fixed in 0.32.9, if I remember correctly.
